wire_protocol/src/connection.rs

```diff
@@ -227,7 +237,13 @@ impl Connection for TlsConnection {
     }
 
     fn is_valid(&self) -> bool {
-        self.stream.as_ref().expect("get connection").is_valid()
+        self.is_valid
+            && self.stream.as_ref().is_some()
```

Comment:
when can `self.stream.as_ref()` be None?

Reply:
the connection can be splited into writeHalf and readHalf. After the split the original stream is None.